news/2016-05-18-enhanced-instance.asciidoc

```diff
@@ -0,0 +1,174 @@
+---
+layout: news
+title: Experiments with enhanced version of javax.enterprise.inject.Instance
+author: Martin Kouba
+priority: 1
+change_frequency: weekly
+excerpt: Weld team is considering adding an enhanced version of javax.enterprise.inject.Instance to the Weld API. Explore the possibilities...
+desc: Experiments with enhanced version of javax.enterprise.inject.Instance
+---
+:linkattrs:
+
+`javax.enterprise.inject.Instance` is sometimes invaluable companion.
+Not only it allows to handle unsatisfied dependency gracefully, i.e. to avoid aborting container initialization unless really necessary:
+
+[source,java]
+----
+interface OrderProcessor {
+  void process(Order order);
+  int getPriority();
+}
+
+class OrderService {
+
+  @Inject
+  Instance<OrderProcessor> instance;
+
+  void create(Order order) {
+    if (!instance.isUnsatisfied() && !instance.isAmbiguous()) {
+        instance.get().process(order);
+    } else {
+        // Log a warning or throw an exception
+    }
+  }
+}
+----
+
+It is less known that `Instance` extends `Iterable` and so it allows to iterate contextual references of beans with the specified combination of required type and qualifiers.
```

Comment:
`to iterate con...` -> `to iterate through con...`
I am missing some kind of preposition here.